_posts/2024-08-24-GSoC2024-dingo.md

```diff
@@ -0,0 +1,230 @@
+
+Copulas are a useful statistical tool for validating significant correlations. However, manually inspecting copula plots can introduce user bias and be time-consuming given the number of paired reactions. An alternative to manual curation is using a metric called the "indicator," which sums the probability values across each copula's diagonal and divides them. The value of the indicator, whether positive or negative, helps validate or reject a possible correlation. Users can set an indicator cutoff that functions similarly to the pearson cutoff.
+
+In addition to calculating correlations, a `plot_corr_matrix` function is available for visualizing the correlation matrix as a heatmap. This visualization helps users easily identify significant correlations.
+
+Code that illustrates how to use these functions:
+```python
+dingo_model = MetabolicNetwork.from_json("ext_data/e_coli_core.json")
+
+sampler = PolytopeSampler(dingo_model)
+steady_states = sampler.generate_steady_states()
+
+corr_matrix, indicator_dict = correlated_reactions(
+                              steady_states,
+                              pearson_cutoff = 0.99,
+                              indicator_cutoff = 2, 
+                              # number of cells to compute the copula
+                              cells = 10,
+                              # value that defines the width of the copula’s diagonal
+                              cop_coeff = 0.3,
+                              # boolean variable that when True, keeps only the lower triangular matrix
+                              lower_triangle = False)
+
+plot_corr_matrix(corr_matrix,
+                 # list containing reactions’ names
+                 reactions,
+                 # parameter that defines image saving format
+                 format = "svg")
+```
+
+The `e_coli_core.json` model we use represents the core metabolism of the E. coli. You can find the model and related data [here](http://bigg.ucsd.edu/models/e_coli_core).
+
+Here’s an example of a heatmap created from a symmetrical correlation matrix without cutoffs:
+<center>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/corr_matrix_no_cutoffs.png"><br><br>
+</center>
+
+We can focus on correlations from reactions that belong to the glycolytic and pentose pathways to get some insights from the heatmap. If you're unfamiliar with these reactions' topology, you can visit [ESCHER](https://escher.github.io/#/) <a href="#ref-3">[3]</a> and load the core metabolism map of E. coli.
+
+We observe strong pairwise correlations among all reactions in the pentose phosphate pathway (`G6PDH2R, PGL, GND, RPE, RPI, TKT1, TALA, TKT2`). However, for the glycolytic pathway (`PGI, PFK, FBA, TPI, GAPD, PGK, PGM, ENO, PYK`), two reactions, `PFK` and `PYK`, show decreased correlation with the others. 
```

Comment:
PFK however has high correlation scores with PGI, TPI and FBA, no ?

Reply:
PFK has a correlation of ~0.81 with PGI, TPI, FBA. The reactions mentioned in the blog have a correlaation of ~0.99.

Reply:
I'd say you mention these numbers indicatively since from the plot it's hard to say the difference. 
It's a real nice example-case.